tuist/tuist
# Model Mappers

(Based on https://github.com/tuist/tuist/pull/1227)

### Short description 📝

Applying modifications to the `Project` and `Target` models within `GraphMappers` may not always be possible due to the current `Graph` structure. To create a mapper that can generate info plists for example, the both the `Target` and the hosting `Project` needs to be modified. 

e.g.

- #1357 
- #1206

### Solution 📦

Add "model mappers" that map `TuistCore` models ahead of converting them to a `Graph` representation.

### Implementation 👩‍💻👨‍💻

_Note: Those can be split into separate PRs to ease the review process but grouped here to demo the concept_

- [x] Add model mappers
- [x] Add recursive manifest loader (loads all manifests without converting them)
- [x] Modify loading process to accommodate model mappers
  - [x] Load all manifests upfront without conversion
  - [x] Lint manifests
  - [x] Convert manifests concurrently to models
  - [x] Apply model mappers

### Test Plan 🛠

- Run `tuist generate` within any of the fixtures
- Verify projects can still be generated as before
- Verify acceptance tests pass
